javascript/0394-decode-string.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Stack
+ * https://leetcode.com/problems/decode-string/
+ * 
+ * Time O(n) | Space O(n)
+ * @param {string} s
+ * @return {string}
+ */
+var decodeString = function(s) {
+  const myStack = [];
+  let result = '';
+
+  for(let i = 0; i < s.length; i++) {
+    if(s[i] !== ']') {
+      myStack.push(s[i]);
+      continue;
+    } 
+
+    let subStr = '';
+    while(myStack[myStack.length - 1] !== '[') {
+      subStr = myStack.pop() + subStr;
+    }
+    myStack.pop();
+
+    let k = '';
+    while(!isNaN(myStack[myStack.length - 1])) {
+      k = myStack.pop() + k;
+    }
+
+    myStack.push(subStr.repeat(+k));
```

Comment:
Repeat will increase time and space

Reply:
Ok, so the new time complexity would be n * (the time it takes to get the number of substrings). The space will be the same right?